src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/AbstractTypeImportCompletionProvider.cs

```diff
@@ -42,9 +42,15 @@ protected override async Task AddCompletionItemsAsync(CompletionContext completi
                 }
                 else
                 {
+                    var isParenthesisCompletionEnable = IsAutoAddParenthesisBySemicolonEnabled(completionContext.Document);
                     foreach (var items in itemsFromAllAssemblies)
                     {
-                        AddItems(items, completionContext, namespacesInScope, telemetryCounter);
+                        AddItems(isParenthesisCompletionEnable
+                            ? AttachParenthesisCompletionProperties(syntaxContext, items).ToImmutableArray()
```

Comment:
We cannot make any changes to the completion items from this provider, those are cached for perf reasons. It's not necessary to do this for all items at this moment, we just need to check for one single item and only when it's being committed, which can be done by getting SyntaxContext in GetChangeAsync.

This is where the cache is:
[http://sourceroslyn.io/#Microsoft.CodeAnalysis.Features/Completion/Providers/ImportCompletionProvider/AbstractTypeImportCompletionService.cs,37](http://sourceroslyn.io/#Microsoft.CodeAnalysis.Features/Completion/Providers/ImportCompletionProvider/AbstractTypeImportCompletionService.cs,37)

This is another cache rely on those CompletionItem to remain unchanged:
[http://sourceroslyn.io/#Microsoft.CodeAnalysis.EditorFeatures/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs,84178c3fc508344c,references](http://sourceroslyn.io/#Microsoft.CodeAnalysis.EditorFeatures/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs,84178c3fc508344c,references)